### PR TITLE
Fix documentation for ELPA_FORCE_REDISTRIBUTE

### PR DESCRIPTION
**Pull request by RMeli**
PR https://github.com/cp2k/cp2k/pull/1515 changed the default value of `ELPA_FORCE_REDISTRIBUTE` from `.FALSE.` to `.TRUE.`, but the corresponding documentation has not been updated.

### CODE REVIEW

File: src/input_cp2k_global.F [hunk 1]
```diff
@@ -723,9 +723,9 @@ SUBROUTINE create_fm_diag_rules_section(section)
 
       CALL keyword_create(keyword, __LOCATION__, name="ELPA_FORCE_REDISTRIBUTE", &
                           description="Controls how to perform redistribution when ELPA is used for diagonalization. "// &
-                          "By default, matrices are redistributed only to prevent crashes in the ELPA library "// &
-                          "which happens when the original matrix is distributed over too many processors. "// &
-                          "By turning on this keyword, redistribution is always performed using the defined rules. ", &
+                          "By default, redistribution is alway performed using the defined rules. "// &
```
Comment (oschuett) [inline src/input_cp2k_global.F]: typo: alway -> always

File: src/input_cp2k_global.F [hunk 2]
```diff
@@ -723,9 +723,9 @@ SUBROUTINE create_fm_diag_rules_section(section)
...
+                          "By turning off this keyword, matrices are redistributed only to prevent crashes in the ELPA"// &
```
Comment (oschuett) [inline src/input_cp2k_global.F]: Missing space after ELPA.

File: src/input_cp2k_global.F (hunk 1 above)
Comment (RMeli) [reply src/input_cp2k_global.F]: Thanks. My editor doesn't like CP2K's formatting and split the line into multiple ones. I ended up removing an extra character by mistake when trying to manually fix things.

File: src/input_cp2k_global.F (hunk 2 above)
Comment (RMeli) [reply src/input_cp2k_global.F]: Yes, thanks. I've seen the CI failure.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -1569,7 +1569,7 @@ target_compile_definitions(
   cp2k
   PUBLIC $<$<BOOL:${CP2K_USE_MPI}>:__parallel>
          $<$<BOOL:${CP2K_USE_MPI}>:__SCALAPACK>
-         $<$<BOOL:${CP2K_USE_F08_MPI}>:__MPI_F08>
+         $<$<BOOL:${CP2K_USE_MPI_F08}>:__MPI_F08>
```
Comment (RMeli) [inline src/CMakeLists.txt]:
```suggestion
         $<$<BOOL:${CP2K_USE_F08_MPI}>:__MPI_F08>
```

File: src/input_cp2k_global.F (hunk 1 above)
Comment (oschuett) [reply src/input_cp2k_global.F]: Out of curiosity, what does your editor use for formatting Fortran code?

Comment (RMeli) [reply src/input_cp2k_global.F]: It uses [fortls](https://github.com/fortran-lang/fortls), but I never really configured it properly (yet). I need to find some time to configure it properly. ;)

**Pull request merged**